lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -1802,9 +1784,22 @@ SubExprInfo ExprEmitter::emitBinary(Operation *op, VerilogPrecedence prec,
 
 SubExprInfo ExprEmitter::emitUnary(Operation *op, const char *syntax,
                                    bool resultAlwaysUnsigned) {
+  // Introduce extra parentheses if `syntax[0]` is emitted immediately before.
+  // This avoids emitting an expression like `a & &b`, which is syntactically
+  // valid but some tool produces LINT warnings.
+
+  // Find a last non-space character in outBuffer.
+  auto it = std::find_if(outBuffer.rbegin(), outBuffer.rend(),
+                         [&](char c) { return c != ' '; });
```

Comment:
This feels pretty janky.

However, the only other solution that comes to mind is to pass have an optional "last char" used that is effectively as robust as the janky solution. What would it look like to also keep track of the last operator character (if one exists) and thread that through?

WDYT?